src/systems/hud/memoize-sprites.js

```diff
@@ -0,0 +1,23 @@
+function readNorm(name, spritesheet) {
+  const frame = spritesheet.frames[name].frame;
+  const size = spritesheet.meta.size;
+  return {
+    x: frame.x / size.w,
+    y: frame.y / size.h,
+    w: frame.w / size.w,
+    h: frame.h / size.h
+  };
+}
+
+export const memoizeSprites = function(spritesheet) {
+  const memo = new Map();
+  return function getSprite(name) {
+    const memoizedSprite = memo.get(name);
+    if (memoizedSprite) {
+      return memoizedSprite;
+    }
+    const sprite = readNorm(name, spritesheet);
+    memo.set(name, sprite);
+    return sprite;
+  };
+};
```

Comment:
If the goal is to establish a mapping from sprite name to `{x, y, w, h}` without creating the `{x, y, w, h}` objects every time, I would have just made an object that is the mapping. The spritesheet isn't very big so doing it lazily doesn't seem important.